streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1176,31 +1176,32 @@ public void cleanUp() {
      * @throws InvalidStateStoreException if Kafka Streams is (re-)initializing or a store with {@code storeName} and
      * {@code queryableStoreType} doesn't exist
      */
+    @Deprecated
     public <T> T store(final String storeName, final QueryableStoreType<T> queryableStoreType) {
-        return store(storeName, queryableStoreType, false);
+        return store(StoreQueryParams.fromNameAndType(storeName, queryableStoreType));
     }
 
     /**
-     * Get a facade wrapping the local {@link StateStore} instances with the provided {@code storeName} if the Store's
+     * Get a facade wrapping the local {@link StateStore} instances with the provided {@link StoreQueryParams}.
+     * StoreQueryParams need required parameters to be set, which are {@code storeName} and if
      * type is accepted by the provided {@link QueryableStoreType#accepts(StateStore) queryableStoreType}.
+     * The optional parameters to the StoreQueryParams include {@code partition} and {@code staleStoresEnabled}.
      * The returned object can be used to query the {@link StateStore} instances.
      *
-     * @param storeName           name of the store to find
-     * @param queryableStoreType  accept only stores that are accepted by {@link QueryableStoreType#accepts(StateStore)}
-     * @param includeStaleStores      If false, only permit queries on the active replica for a partition, and only if the
-     *                            task for that partition is running. I.e., the state store is not a standby replica,
-     *                            and it is not restoring from the changelog.
-     *                            If true, allow queries on standbys and restoring replicas in addition to active ones.
+     * @param storeQueryParams    If StoreQueryParams.fromNameAndType(storeName, queryableStoreType).withPartition(int partition) is used, it allow queries on the specific partition irrespective if it is a standby
+     *                            or a restoring replicas in addition to active ones.
+     *                            If StoreQueryParams.fromNameAndType(storeName, queryableStoreType).withIncludeStaleStores() is used, it allow queries on standbys and restoring replicas in addition to active ones for all the local partitions on the instance.
+     *                            If StoreQueryParams.fromNameAndType(storeName, queryableStoreType).withIncludeStaleStores().withPartition(int partition), it allow queries on the specific partition irrespective if it is a standby
+     *                            or a restoring replicas in addition to active ones..
+     *                            By default, if just storeQueryParams is used, it returns all the local partitions for the store which are in running state.
```

Comment:
Same as above -- we should keep the parameter description brief, as this content belong into the JavaDocs of `StoreQueryParams`

Reply:
Done.

Reply:
Done.